torch/_inductor/codecache.py

```diff
@@ -458,13 +469,17 @@ def _reduce_tensor(t):
     """
     See FxGraphCachePickler. Custom reducer to pickle Tensors.
     """
-    # See GraphLowering. We expect any Tensors to be small inlined constants.
-    # For these, we hash on the metadata _and_ the actual values.
-    assert len(t.shape) == 0 or (len(t.shape) == 1 and t.shape[0] <= 128)
-    metadata_and_values = TensorMetadataAndValues(
-        extract_tensor_metadata(t), t.tolist()
-    )
-    return (_ident, (metadata_and_values,))
+    # If we see tensors, we know they're contstants stored as attributes on
+    # the GraphModule. See tensor lowering; small constants are inlined. If
+    # we see a small tensor, therefore, no reference will ultimately remain
+    # in the generated code. So we need to include its value in the cache key.
+    # Large constannts are effectively treated as inputs and we consider only
+    # their metadata.
+    metadata = extract_tensor_metadata(t)
+    if len(t.shape) == 0 or (len(t.shape) == 1 and t.shape[0] <= 8):
```

Comment:
Can you factor out the logic in torch/_inductor/graph.py about whether or not we are inlining the constants and call that here ? It's fragile to duplicate the logic.

Reply:
@eellison, ah good catch. Thanks; meant to ask about this, but brain farted. I wondered if the logic might already be duplicated because this doesn't look like the place where the inlining actually happens. I found this: https://github.com/pytorch/pytorch/blob/main/torch/_inductor/lowering.py#L2217 and it actually looks slightly different (the check for float or int). Which is actually the better source of truth here?

Reply:
That's a good find! But I don't if that lowering matters since the inputs will be represented in the string of the graph. I think it's just the attr constants that would matter. 

TBh i'm not sure when we would generate `torch.tensor` or `scalar_tensor` - I would have thought those would get replaced by `lift_fresh`. Maybe @ezyang knows since he touched it somewhat recently ? https://github.com/pytorch/pytorch/commit/39ba2e6226eb78ec2e73b0636f6cb599eeff3e62

In any case I'm pretty sure you can ignore that lowering for constant hashing purposes since the input values will be represented in the graph.

Reply:
`scalar_tensor` gets generated if the inputs to torch.tensor have SymInt in them. You can't use lift fresh because the tensor contents are actually dynamic.